job-server/src/main/scala/spark/jobserver/io/FileCacher.scala

```diff
@@ -27,21 +27,45 @@ trait FileCacher {
     appName + "-" + uploadTime.toString("yyyyMMdd_hhmmss_SSS") + s".${binaryType.extension}"
   }
 
+  // Random temporary filename suffix possible chars
```

Comment:
Hi, thanks for the review! Here is my reasoning:

- When using the Java implementation, I cannot control the exact pattern of the file (the "random" part). And I wanted it to have exact, explicitly adjustable length and the character set to choose from.

- When using the Java implementation, I cannot control the exact folder in which the temporary file is stored. Its system-specific. Here we want to control the file location, because of three reasons:

  1. The rename operation is faster and has much less I/O overhead than moving/copying the file itself (which we would have to do, since this would be a different folder than our SJS binaries one)
  2. If the automatic deletion fails, we want to be able find the file easily than to search through the various system tmp directories
  3. If the automatic deletion fails, the FileCacher.cleanCacheBinaries(...) method also will be able to delete the temporary JARs during its normal execution